### PR TITLE
Faster Layout Nodes

### PR DESCRIPTION
**Pull request by Adlai-Holler**
@appleguy @nguyenhuy Been majorly frustrated at work lately so it feels super good to have a nice, clean improvement to offer.

### CODE REVIEW

**Comment by Adlai-Holler** [comment]
Also for sugar on top, I was hoping to use `QOS_CLASS_SOMETHING` to specify the global queue there. The docs for `dispatch_get_global_queue` say `You may specify the values QOS_CLASS_USER_INTERACTIVE, QOS_CLASS_USER_INITIATED, QOS_CLASS_UTILITY, or QOS_CLASS_BACKGROUND.`. This feels like `QOS_CLASS_UTILITY` to me but I want to know what y'all think

**Comment by levi** [comment]
Awesome stuff, @Adlai-Holler!

**Comment by Adlai-Holler** [comment]
Thanks man!

File: AsyncDisplayKit/Details/ASDataController.mm [hunk 1]
```diff
@@ -105,31 +105,27 @@ - (void)_layoutNodes:(NSArray *)nodes atIndexPaths:(NSArray *)indexPaths withAni
...
-        ASCellNode *node = nodes[j + idx];
-        ASSizeRange constrainedSize;
-        [nodeBoundSizes[idx] getValue:&constrainedSize];
+      for (NSUInteger k = j; k < j + batchCount; k++) {
+        ASCellNode *node = nodes[k];
+        ASSizeRange constrainedSize = nodeBoundSizes[k];
         [node measureWithSizeRange:constrainedSize];
-        node.frame = CGRectMake(0.0f, 0.0f, node.calculatedSize.width, node.calculatedSize.height);
-      }];
+        node.frame = CGRectMake(0, 0, node.calculatedSize.width, node.calculatedSize.height);
```
Comment (appleguy) [inline AsyncDisplayKit/Details/ASDataController.mm]: Create a local variable, CGSize calculatedSize = [node measureWithSizeRange:constrainedSize];

Comment (Adlai-Holler) [inline AsyncDisplayKit/Details/ASDataController.mm]: Ah! Good point

**Comment by appleguy** [comment]
Very nice, thank you @Adlai-Holler!  Some of @levi's changes are a possible performance regression in this code, but hopefully we can clean that up in the time before it lands — and having this as a higher bar will inspire us to do so :).

I would suggest we not touch queue priorities for now.  The dispatch headers actually talk about the risk of priority inversion and in at least one place recommend not using priorities unless absolutely necessary.  The main thread already runs at a higher priority, so there is no real risk of dropping frames from this background thread work.  More importantly, if we run at lower priority, it's likely that arbitrary developers' own background tasks (compressing analytics data to send over the network, JSON parsing, etc) could be taking CPU time away from layout, which the user may well be waiting on (at the end of the scrollable area, awaiting for new cells to finish layout and be added).

**Pull request merged**